pkg/trace/event/extractor_single_span_rate.go

```diff
@@ -0,0 +1,40 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package event
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/remoteconfig/client/products/apmsampling"
+	"github.com/DataDog/datadog-agent/pkg/trace/pb"
+	"github.com/DataDog/datadog-agent/pkg/trace/sampler"
+)
+
+// singleSpanRateExtractor is a single span extractor that decides whether to extract spans from trace chunks based on
+// availability of single span sampling tags, e.g. KeySpanSamplingMechanism.
```

Comment:
```suggestion
// singleSpanRateExtractor extracts spans that have been sampled using the single span sampling mechanism.
```
Isn't this more concise? Then, you could go into more detail about what that is on the definition of the `apmsampling.SpanSamplingRuleMechanism` constant.